packages/zoe/src/contractFacet/contractFacet.js

```diff
@@ -46,40 +46,55 @@ export function buildRootObject() {
     /** @type WeakStore<ZCFSeat,ZCFSeatAdmin> */
     const seatToZCFSeatAdmin = makeWeakStore('seat');
 
-    const issuers = Object.values(instanceRecord.issuerKeywordRecord);
+    const keywords = Object.keys(instanceRecord.terms.issuers);
+    const issuers = Object.values(instanceRecord.terms.issuers);
 
     const getPromiseForIssuerRecords = issuersP =>
       Promise.all(issuersP.map(issuerTable.getPromiseForIssuerRecord));
 
-    await getPromiseForIssuerRecords(issuers);
-
-    const allSeatStagings = new WeakSet();
-
     const registerIssuerRecord = (keyword, issuerRecord) => {
-      assertKeywordName(keyword);
-      assert(
-        !getKeywords(instanceRecord.issuerKeywordRecord).includes(keyword),
-        details`keyword ${keyword} must be unique`,
-      );
       instanceRecord = {
         ...instanceRecord,
-        issuerKeywordRecord: {
-          ...instanceRecord.issuerKeywordRecord,
-          [keyword]: issuerRecord.issuer,
-        },
-        brandKeywordRecord: {
-          ...instanceRecord.brandKeywordRecord,
-          [keyword]: issuerRecord.brand,
+        terms: {
+          ...instanceRecord.terms,
+          issuers: {
+            ...instanceRecord.terms.issuers,
+            [keyword]: issuerRecord.issuer,
+          },
+          brands: {
+            ...instanceRecord.terms.brands,
+            [keyword]: issuerRecord.brand,
+          },
+          maths: {
+            ...instanceRecord.terms.maths,
+            [keyword]: issuerRecord.amountMath,
+          },
         },
       };
 
       return issuerRecord;
     };
 
+    const registerIssuerRecordWKeyword = (keyword, issuerRecord) => {
+      assertKeywordName(keyword);
+      assert(
+        !getKeywords(instanceRecord.terms.issuers).includes(keyword),
+        details`keyword ${keyword} must be unique`,
+      );
+      return registerIssuerRecord(keyword, issuerRecord);
+    };
+
+    const issuerRecords = await getPromiseForIssuerRecords(issuers);
+    issuerRecords.forEach((issuerRecord, i) => {
+      registerIssuerRecord(keywords[i], issuerRecord);
```

Comment:
This is the only use of the non-checking form. Why not `registerIssuerRecordWKeyword`?

Reply:
Huh. This loop doesn't correspond to anything in the previous code. Was omitting it an undetected bug that this fixes?

Reply:
Nope, with the addition of amountMath in the terms, we need to overwrite with local amountMath.